### DISCUSSION TITLE
Build versioning information

### DISCUSSION THREAD

**Discussion by dusansimic** [microsoft/openjdk#161] [Q&A] [answered]
Hi! I would like to make a package for a Linux distribution I use and I'd like to have more information about meaning of build versions for Microsoft OpenJDK builds. I can see that besides the OpenJDK versions (17.0.1 or 11.0.13 for example) there is two more numbers (at the time of writing its 12.1 (17.0.1.12.1) and 8.1 (11.0.13.8.1)). What do those numbers represent? I need this information purely for identification of those number in the package specifications.

**Comment by karianna** [comment, marked as answer]
Hi @dusansimic ,

A version string such as 17.0.1.12.1 can be interpreted as follows:

The first 3 numbers are OpenJDK's semantic versioning scheme:

17 - OpenJDK Major Version
0 - OpenJDK Minor Version
1 - OpenJDK Point Version

The 4th number is the OpenJDK build number, for example, the 17.0.1 release was tagged `17.0.1+12` in the OpenJDK repo (we translate that to 17.0.1.12)

The 5th number is Microsoft's build number.  This defaults to 1 but you may see higher numbers if we need to respin a binary.
